staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go

```diff
@@ -123,10 +129,6 @@ func New(lockType string, ns string, name string, coreClient corev1.CoreV1Interf
 		LockConfig: rlc,
 	}
 	switch lockType {
-	case EndpointsResourceLock:
-		return endpointsLock, nil
-	case ConfigMapsResourceLock:
-		return configmapLock, nil
```

Comment:
keep these cases and return specific errors indicating what should be used instead

Reply:
done